Aquality.Appium.Mobile/src/Aquality.Appium.Mobile/Applications/Application.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Aquality.Appium.Mobile.Configurations;
+using Aquality.Selenium.Core.Applications;
+using Aquality.Selenium.Core.Configurations;
+using Aquality.Selenium.Core.Localization;
+using OpenQA.Selenium.Appium;
+using OpenQA.Selenium.Appium.Service;
+using OpenQA.Selenium.Remote;
+using System;
+
+namespace Aquality.Appium.Mobile.Applications
+{
+    public class Application : IMobileApplication
+    {
+        private readonly ILocalizedLogger localizedLogger;
+        private readonly IApplicationProfile applicationProfile;
+        private TimeSpan timeoutImplicit;
+
+        public Application(AppiumDriver<AppiumWebElement> driver, AppiumLocalService driverService = null)
+        {
+            Driver = driver;
+            DriverService = driverService;
+            localizedLogger = AqualityServices.LocalizedLogger;
+            applicationProfile = AqualityServices.ApplicationProfile;
+            SetImplicitlyWaitToDriver(AqualityServices.Get<ITimeoutConfiguration>().Implicit);
+        }
+
+        private void SetImplicitlyWaitToDriver(TimeSpan timeout)
+        {
+            Driver.Manage().Timeouts().ImplicitWait = timeout;
+            timeoutImplicit = timeout;
+        }
+
+        RemoteWebDriver IApplication.Driver => Driver;
+
+        public bool IsStarted => Driver.SessionId != null;
+
+        public AppiumDriver<AppiumWebElement> Driver { get; }
+
+        public AppiumLocalService DriverService { get; }
+
+        public PlatformName PlatformName => applicationProfile.PlatformName;
+
+        public void SetImplicitWaitTimeout(TimeSpan timeout)
+        {
+            if (timeout != timeoutImplicit)
+            {
+                localizedLogger.Debug("loc.application.implicit.timeout", null, timeout.TotalSeconds);
```

Comment:
why do we need 'null' here? are there overrides for Debug with 2 params?

Reply:
yes it is, I'll name optional parameter